crates/pathfinder/src/rpc/types.rs

```diff
@@ -787,36 +850,179 @@ pub mod reply {
 
     /// Used in [Block](crate::rpc::types::reply::Block) when the requested scope of
     /// reply is [BlockResponseScope::FullTransactionsAndReceipts](crate::rpc::types::request::BlockResponseScope).
-    ///
-    /// `contract_address` field is available for Deploy and Invoke transactions.
-    /// `entry_point_selector` and `calldata` fields are available only
-    /// for Invoke transactions.
+    #[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+    #[serde(untagged)]
+    pub enum TransactionAndReceipt {
```

Comment:
This type (and all related code) will be removed in the next PR where we're updating the `starknet_getBlock*` methods. (The new methods are incapable of returning both transactions _and_ receipts.)